### PR TITLE
add ilovbee B87 keyboard

### PR DESCRIPTION
**Pull request by yangzheng20003**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rk/b87/config.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#define CUSTOM_RGBLIGHT_EFFECT_RAINBOW_SWIRL
+
+#define RGBLIGHT_DEFAULT_VAL 60
+#define breathing_val        40
+#define rainbow_mood         40
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_DEFAULT_MODE   14
+#define LED_TYPE                RGB
+/* WS2812 */
+#define WS2812_SPI_DRIVER       SPIDM2
```
Comment (itarze) [inline keyboards/rk/b87/config.h]:
```suggestion
#define LED_TYPE                RGB

/* WS2812 */
#define WS2812_SPI_DRIVER       SPIDM2
```

File: keyboards/rk/b87/config.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#define RGBLIGHT_EFFECT_BREATHING
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_TWINKLE
+#define RGBLIGHT_DISABLE_KEYCODES
```
Comment (itarze) [inline keyboards/rk/b87/config.h]: Move to keyboard.json

File: keyboards/rk/b87/config.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+#define CUSTOM_RGBLIGHT_EFFECT_BREATHE_TABLE
+#define CUSTOM_RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define CUSTOM_RGBLIGHT_EFFECT_RAINBOW_SWIRL
+
+#define RGBLIGHT_DEFAULT_VAL 60
```
Comment (itarze) [inline keyboards/rk/b87/config.h]: Move to keyboard.json

File: keyboards/rk/b87/rgb_record/rgb_record.c [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+    for (uint8_t i = 0; i < (sizeof(sixth_gear_buff) / sizeof(sixth_gear_buff[0])); i++) {
+        if (rgb_matrix_get_mode() == sixth_gear_buff[i]) {
+            temp = RGB_HSV_MAX - 1;
+            break;
+        } else if (i == (sizeof(sixth_gear_buff) / sizeof(sixth_gear_buff[0]) - 1)) {
+            temp = RGB_HSV_MAX;
+        }
+    }
+    if (status) rgb_hsv_index = (rgb_hsv_index + 1) % temp;
```
Comment (itarze) [inline keyboards/rk/b87/rgb_record/rgb_record.c]:
```suggestion
    }
    
    if (status) rgb_hsv_index = (rgb_hsv_index + 1) % temp;
```

File: keyboards/rk/b87/rgb_record/rgb_record.mk [hunk 5]
```diff
@@ -0,0 +1,2 @@
+SRC += rgb_record/rgb_record.c
+SRC += rgb_record/rgb_rgblight.c
```
Comment (itarze) [inline keyboards/rk/b87/rgb_record/rgb_record.mk]:
```suggestion
SRC += rgb_record/rgb_rgblight.c

```

**Review by itarze — CHANGES_REQUESTED** [review]
rgblight.c formats only your changes.

**Comment by Kessmex** [comment]
Would appreciate any follow up on this! Been trying to make this keyboard work for weeks now however I have no coding knowledge and I can't make much by myself

**Pull request closed without merging**